docs/plugin.md

```diff
@@ -0,0 +1,97 @@
+To add custom rules first extend the [Rule](https://github.com/Skyscanner/cfripper/blob/master/cfripper/model/rule.py)
+class. Then implement the `invoke` method by adding your logic.
+
+{{ inline_source('cfripper.rules.base_rules.Rule.invoke') }}
+
+CFRipper uses [pycfmodel](https://github.com/Skyscanner/pycfmodel) to create a Python model of the CloudFormation script.
+This model is passed to the `invoke` function as the `cfmodel` parameter. You can use the model's iterate through the
+resources and other objects of the model and use the helper functions to perform various checks. Look at the
+[current rules](/rules) for examples.
+
+## Making your rules available to CFRipper
+
+CFRipper in order to load dynamic rules uses [pluggy](https://github.com/pytest-dev/pluggy). You have to define a 
+so-called entry point for your distribution so that cfripper finds your plugin module. Entry points are a feature 
+provided by setuptools. CFRipper CLI looks up the `cfripper` entrypoint to discover its plugins and load them.
+
+CFRipper just have one hook that has to be implemented to provide custom rules. `cfripper_get_rules` hook returns a 
+dictionary where key are strings and values are rule inherited classes.
+
+### Example
+
+This is a two file example (setup.py and plugin.py) than provides one custom rule to CFRipper. 
+
+setup.py
+```python
+from setuptools import setup
+
+
+setup(
+    name="cfripper-shiny-new-plugin",
+    install_requires="cfripper>=0.24.0,<0.25.0",
+    # the following makes a plugin available to cfripper
+    entry_points={"cfripper": ["shiny_new_plugin = plugin"]},
```

Comment:
Do we need to define and record the current behaviour around what happens if multiple entrypoints define custom rules with the same name or the name of an existing rule? Which one will be used (or are they all invoked)?

Reply:
In this case, the last one loaded will be preserved.

Reply:
Thanks for checking! It might be good to add that to documentation at one point just so we know in the future 👍

Reply:
Yep, I'm adding it to docs